src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs

```diff
@@ -311,9 +311,7 @@ public bool Equals(ValueTuple<T1> other)
 
         bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer)
         {
-            if (other == null || !(other is ValueTuple<T1>)) return false;
-
-            var objTuple = (ValueTuple<T1>)other;
+            if (other == null || other is not ValueTuple<T1> objTuple) return false;
 
             return comparer.Equals(Item1, objTuple.Item1);
```

Comment:
Can this just be:
```C#
other is ValueTuple<T1> objTuple && comparer.Equals(Item1, objTuple.Item1)
```
?

Reply:
It could. Do we mind if the code looks slightly different compared to arities > 1?

Reply:
Can't the higher arities be similar? e.g.
```C#
bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer) =>
    other is ValueTuple<T1> vt &&
    comparer.Equals(Item1, objTuple.Item1);

bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer) =>
    other is ValueTuple<T1, T2> vt &&
    comparer.Equals(Item1, objTuple.Item1) &&
    comparer.Equals(Item2, objTuple.Item2);

bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer) =>
    other is ValueTuple<T1, T2, T3> vt &&
    comparer.Equals(Item1, objTuple.Item1) &&
    comparer.Equals(Item2, objTuple.Item2) &&
    comparer.Equals(Item3, objTuple.Item3);

... // etc.
```
?

Reply:
Fair enough, addressed.